MaterialLib/PorousMedium/UnsaturatedProperty/RelativePermeability/CreateRelativePermeabilityModel.cpp

```diff
@@ -0,0 +1,227 @@
+*/
+std::unique_ptr<RelativePermeability> createNonWettingPhaseBrookCoreyOilGas(
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__type}
+    config.checkConfigParameter("type", "NonWettingPhaseBrookCoreyOilGas");
+
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__NonWettingPhaseBrookCoreyOilGas__sr}
+    const double Sr = config.getConfigParameter<double>("sr");
+
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__NonWettingPhaseBrookCoreyOilGas__smax}
+    const double Smax = config.getConfigParameter<double>("smax");
+
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__NonWettingPhaseBrookCoreyOilGas__m}
+    const double m = config.getConfigParameter<double>("m");
+    if (m < 1.0)  // m >= 1
+    {
+        OGS_FATAL(
+            "The exponent parameter of NonWettingPhaseBrookCoreyOilGas\n"
+            "relative permeability model, m, must not be smaller than 1");
+    }
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__NonWettingPhaseBrookCoreyOilGas__m}
+    const double krel_max = config.getConfigParameter<double>("krel_min");
+
+    return std::unique_ptr<RelativePermeability>(
+        new NonWettingPhaseBrookCoreyOilGas(Sr, Smax, m, krel_max));
+}
+
+std::unique_ptr<RelativePermeability> createRelativePermeabilityModel(
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__type}
+    auto const type = config.peekConfigParameter<std::string>("type");
+
+    if (type == "WettingPhaseVanGenuchten")
+    {
+        return createWettingPhaseVanGenuchten(config);
+    }
+    else if (type == "NonWettingPhaseVanGenuchten")
+    {
+        return createNonWettingPhaseVanGenuchten(config);
+    }
+    else if (type == "WettingPhaseBrookCoreyOilGas")
+    {
+        return createWettingPhaseBrookCoreyOilGas(config);
+    }
+    else if (type == "NonWettingPhaseBrookCoreyOilGas")
+    {
+        return createNonWettingPhaseBrookCoreyOilGas(config);
+    }
+    else if (type == "Curve")
+    {
+        config.checkConfigParameter("type", "Curve");
+
+        std::vector<double> variables, values;
+        //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__Curve__curve}
+        auto const& curve_config = config.getConfigSubtree("curve");
+        for (
+            auto const& data_string :
+            //! \ogs_file_param{material_property__porous_medium__porous_medium__relative_permeability__Curve_curve__data}
+            curve_config.getConfigParameterList<std::string>("data"))
+        {
+            std::stringstream ss(data_string);
+            double var, val;
+            ss >> var >> val;
```

Comment:
Thank you for the change. I misunderstood your previous code. Sadly, currently we cannot get around `<point><data> 0. 0.9   </data></point>`, because, there is no `std::vector<T>` overload for `ConfigTree::getValue<T>()`.